### DISCUSSION TITLE
電腦突然無法LISTEN 常用的4200 與3000 通訊埠

### DISCUSSION THREAD

**Discussion by LinChloe** [1101-GS4719-C/Python-Community-GS4719-C#1945] [General]
最近在開啟docker時，發生了點小問題。
當我開啟docker後，想進入workspace容器裡，於是下了指令。
`docker-compose exec workspace bash`
但卻沒有我想像中成功進入，對方回應了我。
`docker: Error response from daemon: Ports are not available: listen tcp 0.0.0.0:4449: bind: An attempt was made to access a socket in a way forbidden by its access permission`

這時你上網查資料或問chatgpt時，它們都會讓你去查此port是否已被佔用。
如果是的話，就照著它們的做法繼續下去吧 !
**但當你去CMD輸入查看port指令，發現根本就沒人在使用時。**
那你可能發生了和我一樣的狀況🥲

這稱為由「**WIN10產生的靈異事件**」

解決方案:

輸入於terminal
`net stop winnat`
`netsh int ipv4 delete excludedportrange protocol=tcp numberofports=1 startport=8888`

_numberofports=”要改的數量”
startport=”要改的起始位置”_

ex: 4123~4153
numberofports=30
startport=4123

網上找到的原始解決辦法，讓我們一起謝謝這位大大~
[https://blog.miniasp.com/post/2019/03/31/Ports-blocked-by-Windows-10-for-unknown-reason](url)

請善心人士幫我按讚、留言、加分享 👍

**Comment by SusuTima** [comment]
感謝分享。我在遇到操作相關的問題時，大部分時候會先複製到網路上去尋找error message 對應的解決方式，只有在完全看不懂網路上給的解決辦法時才會尋求chatgpt的幫助，chatgpt的好處就是他會幫你翻譯對應的解決方式，也會解釋error message出現的原因，但是習慣上還是會先上網自己找解決方式。
